app/Http/Resources/CastResource.php

```diff
@@ -17,7 +18,7 @@ public function toArray($request)
         return [
             'id' => $this->id,
             'medium_id' => $this->medium_id,
-            'person_id' => $this->person_id,
+            'person_data' => new PersonResource(Person::findOrFail($this->id)),
```

Comment:
Wrong! You are looking for a person with a role id, not a person id!!!
Additionally in the cast model you have the person() function which allows you to get exactly this person without giving any keys. (there is no possibility of a mistake you inadvertently made)
Finally it should look like this:
'person' => new PersonResource($this->person),